playwright/tests/common/environment.spec.ts

```diff
@@ -0,0 +1,205 @@
+            path: "prismicRepository.fetchEnvironments",
+            data: () => ({ environments }),
+            execute: false,
+          },
+        ],
+      });
+
+      await sliceMachinePage.gotoDefaultPage();
+      await sliceMachinePage.environmentSelector.selectEnvironment(
+        environments[0].name,
+      );
+    },
+  );
+
+  test.run()(
+    "I don't see login text or the select if I'm not authorized",
+    async ({ sliceMachinePage }) => {
+      await mockManagerProcedures({
+        page: sliceMachinePage.page,
+        procedures: [
+          {
+            path: "prismicRepository.fetchEnvironments",
+            data: () => ({
+              error: { name: "UnauthorizedError" },
+            }),
+            execute: false,
+          },
+        ],
+      });
+
+      await sliceMachinePage.gotoDefaultPage();
+      await expect(
+        sliceMachinePage.environmentSelector.loginTextButton,
+      ).not.toBeVisible();
+      await expect(
+        sliceMachinePage.environmentSelector.loginIconButton,
+      ).not.toBeVisible();
+      await expect(
+        sliceMachinePage.environmentSelector.dropdownTrigger,
+      ).not.toBeVisible();
+    },
+  );
+
+  test.run()(
+    "I can see the dot on the logo depending on the environment",
+    async ({ sliceMachinePage }) => {
+      await mockManagerProcedures({
+        page: sliceMachinePage.page,
+        procedures: [
+          {
+            path: "prismicRepository.fetchEnvironments",
+            data: () => ({ environments }),
+            execute: false,
+          },
+        ],
+      });
+
+      await sliceMachinePage.gotoDefaultPage();
+
+      await sliceMachinePage.environmentSelector.selectEnvironment(
+        environments[0].name,
+      );
+      await sliceMachinePage.environmentSelector.checkSelectedEnvironmentKind(
+        environments[0].kind,
+      );
+
+      await sliceMachinePage.environmentSelector.selectEnvironment(
+        environments[1].name,
+      );
+      await sliceMachinePage.environmentSelector.checkSelectedEnvironmentKind(
+        environments[1].kind,
+      );
+
+      await sliceMachinePage.environmentSelector.selectEnvironment(
+        environments[2].name,
+      );
+      await sliceMachinePage.environmentSelector.checkSelectedEnvironmentKind(
+        environments[2].kind,
+      );
+    },
+  );
+
+  test.run()(
+    "I can see the window top border depending on the environment",
+    async ({ sliceMachinePage }) => {
+      await mockManagerProcedures({
+        page: sliceMachinePage.page,
+        procedures: [
+          {
+            path: "prismicRepository.fetchEnvironments",
+            data: () => ({ environments }),
+            execute: false,
+          },
+        ],
+      });
+
+      await sliceMachinePage.gotoDefaultPage();
+
+      await sliceMachinePage.environmentSelector.selectEnvironment(
+        environments[0].name,
+      );
+      await expect(sliceMachinePage.topBorder).toHaveCSS(
```

Comment:
⚠️ I feel like it's a bit of implementation details here. Can we not have a role as you did for the dot with an invisible text? I feel it would be the same here as the dot. WDYT?

Reply:
We could change the border and test to use `role="status"`, but then we wouldn't be testing the color. Instead, we would be testing the status.

I don't think we should mark the top border as a status item, since we already have another status indicator providing the same information (the dot). Thus, we would be adding text purely for testing purposes since it would not be announced.

In this case, would be make more sense to use a computed `data-cy` (e.g. `data-cy="top-border-prod"`) or `data-environment-kind="prod"`? We don't check the color of the dot either, so maybe we should just check for kind instead.

Reply:
Honestly, I guess for this first test it's ok. Product is unsure of this part, so if one day we change the color or something like this we could make the improvement at the same time. WDYT?

Reply:
Works for me 👍